tests/foreman/cli/test_http_proxy.py

```diff
@@ -85,33 +85,55 @@ def test_positive_create_update_delete(module_org, module_location, target_sat):
 
 
 @pytest.mark.tier3
-@pytest.mark.run_in_one_thread
-@pytest.mark.stubbed
-def test_insights_client_registration_with_http_proxy():
+@pytest.mark.no_containers
+@pytest.mark.rhel_ver_match(r'^(?!6$)\d+$')
+@pytest.mark.parametrize(
+    'setup_http_proxy',
+    [True, False],
+    indirect=True,
+    ids=['auth_http_proxy', 'unauth_http_proxy'],
+)
+def test_insights_client_registration_with_http_proxy(
+    module_target_sat,
+    setup_http_proxy,
+    rhel_contenthost,
+    rhcloud_activation_key,
+    rhcloud_manifest_org,
+):
     """Verify that insights-client registration work with http proxy.
 
     :id: 5158d5c1-2b88-4c05-914b-f1c53656ffc2
 
-    :customerscenario: true
+    :parametrized: yes
 
-    :steps:
-        1. Create HTTP Proxy.
-        2. Set created proxy as "Default HTTP Proxy" in settings.
-        3. Edit /etc/resolv.conf and comment out all entries so that
-            satellite can not directly communicate outside. Ensure that
-            NetworkManger won't change it.
-        4. Register a host with satellite.
-        5. Register host with insights.
-        6. Try insights-client register/unregister/test-connection/status
+    :setup:
+        1. Satellite with Default HTTP Proxy set.
 
-    :BZ: 1959932
+    :steps:
+        1. Register a host with satellite.
+        2. Register host with insights.
+        3. Try insights-client register/unregister/test-connection/status
 
     :expectedresults:
-        1. insights-client register/unregister/test-connection/status
-            works with http proxy set.
+        1. insights-client register/unregister/test-connection/status works with http proxy set.
 
-    :CaseAutomation: NotAutomated
+    :BZ: 1959932
+
+    :customerscenario: true
     """
+    rhel_contenthost.configure_rex(
+        satellite=module_target_sat, org=rhcloud_manifest_org, register=False
+    )
+    rhel_contenthost.configure_rhai_client(
+        satellite=module_target_sat,
+        activation_key=rhcloud_activation_key.name,
+        org=rhcloud_manifest_org.label,
+        rhel_distro=f"rhel{rhel_contenthost.os_version.major}",
+    )
+    assert rhel_contenthost.execute('insights-client --register').status == 0
```

Comment:
configure_rhai_client() will register this client to satellite, so here are we checking if it could be registered again to same satelllite? if not, can we remove this assertion?

Reply:
This is correct, I think testing re-register won't harm.